### PR TITLE
V1

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: conf/default.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+plugins:
+  hapi-upload-s3:
+    endpoint: '/api/media'
+    s3Bucket: '{{ENV.MEDIAMANAGER_BUCKET}}'
+    s3SecretAccessKey: '{{ENV.MEDIAMANAGER_SECRETKEY}}'
```
Comment (jgallen23) [inline conf/default.yaml]: lets move the s3 stuff out into its own object and then reference them here.

File: methods/get.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const AWS = require('aws-sdk');
+const s3 = new AWS.S3({ apiVersion: '2006-03-01' });
+
+module.exports = function(key, callback) {
+  const options = this.registrations['hapi-upload-s3'].options;
```
Comment (jgallen23) [inline methods/get.js]: switch this to use `this.settings.app.s3...`

File: methods/get.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const AWS = require('aws-sdk');
```
Comment (jgallen23) [inline methods/get.js:1]: what does this method do?  return the image content?

File: routes/test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+module.exports.dummy = {
+  path: '/test',
+  method: 'GET',
+  handler: (request, reply) => {
+    reply(`
```
Comment (jgallen23) [inline routes/test.js]: can we switch this to use vision with handlebars.  We'll need this later anyways

File: methods/get.js (hunk 3 above)
Comment (orthagonal) [reply methods/get.js:1]: Yeah, it returns the raw image data.  If it was uploaded with an s3 key name ending in .jpg/.png/.etc, your browser will interpret it as a valid image file and download it.

File: methods/get.js [hunk 5]
```diff
@@ -2,11 +2,9 @@ const AWS = require('aws-sdk');
 const s3 = new AWS.S3({ apiVersion: '2006-03-01' });
 
 module.exports = function(key, callback) {
-  const options = this.registrations['hapi-upload-s3'].options;
-  const s3Options = {};
-  s3Options.Bucket = options.s3Bucket;
-  s3Options.Key = key;
-  s3.getObject(s3Options, (err, data) => {
+  const options = this.settings.app.s3;
```
Comment (jgallen23) [inline methods/get.js]: you probably want to clone this, otherwise `Key` is set on the settings object

File: conf/default.yaml [hunk 6]
```diff
@@ -1,8 +1,20 @@
+debug: true
+s3:
+  Bucket: '{{ENV.MEDIAMANAGER_BUCKET}}'
+  SecretAccessKey: '{{ENV.MEDIAMANAGER_SECRETKEY}}'
+  AccessKeyId: '{{ENV.MEDIAMANAGER_ACCESSKEY}}'
+  Region: 'us-east-1'
```
Comment (jgallen23) [inline conf/default.yaml]: this should be an env var too

File: conf/default.yaml [hunk 7]
```diff
@@ -1,8 +1,20 @@
+debug: true
+s3:
+  Bucket: '{{ENV.MEDIAMANAGER_BUCKET}}'
```
Comment (jgallen23) [inline conf/default.yaml]: don't need `MEDIAMANAGER_` prefix

File: conf/default.yaml (hunk 6 above)
Comment (orthagonal) [reply conf/default.yaml]: Confi is actually throwing a yaml parse error when I do {{s3.region}} there, not sure why.  (fixed)

Comment (jgallen23) [reply conf/default.yaml]: because it can't reference itself

File: conf/default.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+debug: true
+s3:
+  Bucket: '{{ENV.BUCKET}}'
```
Comment (jgallen23) [inline conf/default.yaml]: we typically don't capitalize key names

File: conf/default.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  SecretAccessKey: '{{ENV.SECRETKEY}}'
```
Comment (jgallen23) [inline conf/default.yaml]: we should use the aws versions. AWS_SECRET_KEY (I think)

File: conf/default.yaml [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+views:
+  engines:
+    html: 'handlebars'
+  path: 'public'
+plugins:
+  vision:
+    _priority: -1
+  hapi-upload-s3:
+    endpoint: '/api/media'
+    s3Bucket: '{{s3.bucket}}'
```
Comment (jgallen23) [inline conf/default.yaml:16]: these won't work. don't match the key names

File: conf/default.yaml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    s3SecretAccessKey: '{{s3.secretaccesskey}}'
```
Comment (jgallen23) [inline conf/default.yaml]: same as above, this is all lowercase, so won't work

**Comment by jgallen23** [comment]
make sure you are testing these things. I shouldn't be able to see errors by just looking at the code

File: conf/default.yaml (hunk 8 above)
Comment (orthagonal) [reply conf/default.yaml]: These are reserved case-sensitive keynames used by aws-sdk. If we lowercase them in default.yaml we will still have to uppercase them when the method is called, before we can pass them to AWS.  I'm cool with it either way, let me know if you still want me to change this @jgallen23

**Pull request merged**